src/main/java/jimmy/mcgymmy/logic/predicate/NameContainsKeywordsPredicate.java

```diff
@@ -22,13 +23,14 @@ public NameContainsKeywordsPredicate(List<String> keywords) {
     @Override
     public boolean test(Food food) {
         assert food != null : "NameContainsKeywordsPredicate -> Name cannot be null";
-        try {
-            return keywords.stream()
-                    .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(food.getName().fullName, keyword));
-        } catch (IllegalArgumentException e) {
-            logger.info("ILE surfaced : " + e);
-            return true;
-        }
+        return keywords.stream()
+                .anyMatch(keyword -> {
+                    try {
+                        return StringUtil.containsWordIgnoreCase(food.getName().fullName, keyword);
+                    } catch (IllegalValueException e) {
+                        return false;
+                    }
```

Comment:
By changing boolean value upon catching from true to false, does this change the behaviour as specified in UG?

Reply:
have to ask @Jh123x about this

Reply:
ok just checked, this implementation is correct. The exception is thrown for invalid inputs (empty string, etc).
These inputs should not even show up because of the way our parser works, but if they do, we want them to not match any of the food items.

Reply:
Great!